modules/ipp/ipp.go

```diff
@@ -0,0 +1,76 @@
+package ipp
+
+import (
+	"bytes"
+	"encoding/binary"
+	//"io"
+	"net"
+)
+
+type Connection struct {
+	Conn net.Conn
+}
+
+//func ReadResponse(body *io.ReadCloser) *ScanResults {
+//	result := &ScanResults{}
+//
+//}
+
+// Returns a byte-encoded "attribute-with-one-value" with the provided "value-tag", "name", and "value"
+// attribute-with-one-value encoding described at https://tools.ietf.org/html/rfc8010#section-3.1.4
+// Example (runnable from ipp_test.go):
+//   Input: 0x47, "attributes-charset", "us-ascii"
+//   Output: [71 0 18 97 116 116 114 105 98 117 116 101 115 45 99 104 97 114 115 101 116 0 8 117 115 45 97 115 99 105 105]
+// TODO: Should return an error when fed an invalid valueTag?
+// TODO: Determine whether this should remain public. Currently is for Testable Example
+func AttributeByteString(valueTag byte, name string, value string) []byte {
+	//special byte denoting value syntax
+	b := []byte{valueTag}
```

Comment:
It looks like this could be simplified by making a single `bytes.Buffer` and writing to that everywhere, instead of going back and forth `append`ing to `b`